pkg/controller/route/route_controller.go

```diff
@@ -177,21 +180,21 @@ func (rc *RouteController) reconcile(nodes []*v1.Node, routes []*cloudprovider.R
 					rc.updateNetworkingCondition(nodeName, err == nil)
 					if err != nil {
 						msg := fmt.Sprintf("Could not create route %s %s for node %s after %v: %v", nameHint, route.DestinationCIDR, nodeName, time.Now().Sub(startTime), err)
-						if rc.recorder != nil {
-							rc.recorder.Eventf(
-								&v1.ObjectReference{
-									Kind:      "Node",
-									Name:      string(nodeName),
-									UID:       types.UID(nodeName),
-									Namespace: "",
-								}, v1.EventTypeWarning, "FailedToCreateRoute", msg)
-						}
-						glog.Error(msg)
-
-					} else {
-						glog.Infof("Created route for node %s %s with hint %s after %v", nodeName, route.DestinationCIDR, nameHint, time.Now().Sub(startTime))
-						return
+						rc.recorder.Eventf(
```

Comment:
Why did you remove the "rc.recorder != nil" if?

Reply:
It can never be nil, see pkg/controller/route/route_controller.go#L80

Reply:
If you're creating it New method.
But you can create it on your own e.g. for test purposes and then it can potentially be nil.

Please revert this if.

Reply:
ok

Reply:
@wojtek-t  done